### DISCUSSION TITLE
Getting it to work behind reverse proxy

### DISCUSSION THREAD

**Discussion by MenooMeow** [inventree/InvenTree#8648] [General]
I'm using 2 different computers at my house.  Both are behind a router.  One computer has an NGINX reverse proxy that routes the various domains to the appropriate Virtual Machine.  I set up the reverse proxy and the DNS so that Inventree is available at  https://inventree.example.com/    I used a Let's Encrypt certificate at the NGINX reverse proxy.  For everything else I do (NextCloud, php sites, Wordpress), this works perfectly.  But, when I try to get Inventree working this way, the browser gives the error:  "
```
inventree.example.com redirected you too many times   
Try deleting your cookies.
ERR_TOO_MANY_REDIRECTS
```
".  

I did update the .env file to reference  "https://inventree.example.com" under the INVENTREE_SITE_URL parameter.

Is there something else I need to do to get it working behind an externally hosted NGINX reverse proxy?

UPDATE 1:
I edited .env --> INVENTREE_SITE_URL="http://inventree.example.com" to remove the S from https.  Now I can access the site through the internet by going to http://inventree.example.com and it seems to redirect to https on its own.  I assume the container is somehow doing some kind of certificate magic?  Is this a secure enough configuration or should I be able to get to https://inventree.example.com without it first starting at http://inventree.example.com?  

Also, it might be related but I cannot log in with the admin username/password I specified in the .env file.  It says:
```
Authentication Failure
You have been logged out from InvenTree.
```
### UPDATE 2: 
**------------- SOLUTION SO FAR -----------------**
I added the following string to the end of the .env file and it seems to work now:
```
INVENTREE_SITE_URL="http://inventree.example.com"
INVENTREE_TRUSTED_ORIGINS="https://inventree.example.com"
```
Maybe the documentation should include instructions on how to work in an external reverse proxy since it's not an uncommon configuration?

**Comment by Jetro97** [comment]
This solved my problem. Sure it would be helpful to include in clear instructions for home users which proxy everything out a single IP.

Many thanks

**Comment by LouisCassany** [comment]
That solved my issue as well and I was able to run inventree with the docker-compose stack while still being able to use my original nginx setup without any port conflicts or redirect loop.
